monai/data/dataset.py

```diff
@@ -878,6 +879,127 @@ def _transform(self, index: int):
+    `Randomizable` `Transform` within a `Compose` instance.
+    So to improve the caching efficiency, please always put as many as possible non-random transforms
+    before the randomized ones when composing the chain of transforms.
+
+    For example, if the transform is a `Compose` of::
+
+        transforms = Compose([
+            LoadImaged(),
+            EnsureChannelFirstd(),
+            Spacingd(),
+            Orientationd(),
+            ScaleIntensityRanged(),
+            RandCropByPosNegLabeld(),
+            ToTensord()
+        ])
+
+    when `transforms` is used in a multi-epoch training pipeline, before the first training epoch,
+    this dataset will cache the results up to ``ScaleIntensityRanged``, as
+    all non-random transforms `LoadImaged`, `EnsureChannelFirstd`, `Spacingd`, `Orientationd`, `ScaleIntensityRanged`
+    can be cached. During training, the dataset will load the cached results and run
+    ``RandCropByPosNegLabeld`` and ``ToTensord``, as ``RandCropByPosNegLabeld`` is a randomized transform
+    and the outcome not cached.
+
+    Note:
+        Unlike `CacheDataset` class, `SharedCacheDataset` caches the full dataset in the shared memory (via ListProxy),
+        so each process within DistributedDataParallel can access items previously cached by other processes.
+        Data is cached on the fly, so there is no need to wait to cache all the data beforehand.
+    """
+
+    def __init__(
+        self,
+        data: Sequence,
+        transform: Optional[Union[Sequence[Callable], Callable]] = None,
+        copy_cache: bool = False,
+        as_contiguous: bool = True,
+        cache_list: Optional[ListProxy] = None,
+        use_cache: bool = True,
+    ) -> None:
+        """
+        Args:
+            data: input data to load and transform to generate dataset for model.
+            transform: transforms to execute operations on input data.
+            copy_cache: whether to `deepcopy` the cache content before applying the random transforms,
+                default to `False`.
+            as_contiguous: whether to convert the cached NumPy array or PyTorch tensor to be contiguous.
+                it may help improve the performance of following logic.
+            cache_list: ListProxy instance created by a master process, to hold the shared memory list
+                It must be crated on the master process e.g. as cache_list = multiprocessing.Manager().list()
+                before spawning/forking the subprocesses. If no cache_list is provided, a local (per process)
+                non-shared cache list will be created (could be sufficient in single gpu environment)
+            use_cache: whether to use cache mechanism. Defaults to True.  When False, the logic becomes equivalent
+                to a Dataset super class (no caching)
+        """
+        if not isinstance(transform, Compose):
+            transform = Compose(transform)
```

Comment:
If `transform` is not a sequence an exception should be raised here.

Reply:
sounds good. thank you